utils/api/fetchProofOfYieldByDay.ts

```
@@ -0,0 +1,85 @@
+            apy
+            fees
+            amoSupply
+            totalSupply
+            nonRebasingSupply
+          }
+          rebases(orderBy: timestamp_DESC, where: {timestamp_gte: "${timestamp}T00:00:00.000Z", timestamp_lt: "${oneDayLater}T00:00:00.000Z"}) {
+            blockNumber
+            fee
+            totalSupply
+            txHash
+            yield
+            timestamp
+          }
+        }`,
+        variables: null,
+        operationName: "ProofOfYieldByDay",
+      }),
+    });
+    const json = await res.json();
+
+    const item = json.data.proofOfYieldById;
+    const rebaseEvents = json.data.rebases;
+
+    const rawApr =
+      (Number(BigInt(item.totalSupply) - BigInt(item.nonRebasingSupply)) /
+        Number(BigInt(item.totalSupply) - BigInt(item.amoSupply))) *
+      item.apy;
+
+    const rawApy = ((1 + rawApr / 365.25 / 100) ** 365.25 - 1) * 100;
+
+    const apyBoost =
+      Number(BigInt(item.totalSupply) - BigInt(item.amoSupply)) /
+      Number(BigInt(item.totalSupply) - BigInt(item.nonRebasingSupply));
+
+    return [
+      {
+        date: item.timestamp,
+        yield: formatEther(item.yield),
+        fees: formatEther(item.fees),
+        backing_supply: formatEther(item.totalSupply),
+        rebasing_supply: formatEther(item.rebasingSupply),
+        non_rebasing_supply: formatEther(item.nonRebasingSupply),
+        apy: item.apy.toString(),
+        raw_apy: rawApy.toFixed(3),
+        apy_boost: apyBoost.toFixed(3),
+        rebase_events: rebaseEvents.map((event) => {
+          const amount = BigInt(event.yield) - BigInt(event.fee);
+          return {
+            amount: Number(formatEther(amount)),
+            fee: Number(formatEther(event.fee)),
+            tx_hash: event.txHash,
+            block_number: event.blockNumber,
+            block_time: event.timestamp,
+          };
+        }),
+      },
+    ];
+  } catch (err) {
+    console.log(`Failed to fetch daily stats: ${err}`);
+  }
```

What happens when we error in these? How does the front-end react? It seems like a bad idea to silently fail. I wouldn't do this. I'd put any silence on failures in the view component.

---

the old function was in this format, I just changed the data fetching piece. I think there are a lot of things like this but I didn't want to start refactoring before we decide the longer term future of the marketing sites